listenbrainz/db/pinned_recording.py

```diff
@@ -40,28 +41,37 @@ def unpin(user_id: int):
 
         Args:
             user_id: the row ID of the user in the DB
+
+        Returns:
+            True if an active pinned recording was unpinned,
+            False if no pinned recording belonging to the given user_id was currently pinned.
     """
 
     with db.engine.connect() as connection:
-        connection.execute(sqlalchemy.text("""
+        result = connection.execute(sqlalchemy.text("""
             UPDATE pinned_recording
                SET pinned_until = NOW()
              WHERE (user_id = :user_id AND pinned_until >= NOW())
             """), {
             'user_id': user_id,
             }
         )
+        return result.rowcount == 1
```

Comment:
While its unlikely the user will have more than one pin at a time (unless something is wrong on our end), I think its still better do this check better as `result.rowcount > 0`.